File: resq_app/lib/models/bill.dart
class Bill {
  final int billId;
  final double servicePrice;
  final double distancePrice;
  final double extraPrice;
  final double totalPrice;
  final double appFee;
  final double discountAmount;
  final double total;
  final String method;
  final String status;
  final String currency;

  Bill({
    required this.billId,
    required this.servicePrice,
    required this.distancePrice,
    required this.extraPrice,
    required this.totalPrice,
    required this.appFee,
    required this.discountAmount,
    required this.total,
    required this.method,
    required this.status,
    required this.currency,
  });

  factory Bill.fromJson(Map<String, dynamic> json) {
    return Bill(
      billId: json['billId'],
      servicePrice: json['servicePrice']?.toDouble() ?? 0.0,
      distancePrice: json['distancePrice']?.toDouble() ?? 0.0,
      extraPrice: json['extraPrice']?.toDouble() ?? 0.0,
      totalPrice: json['totalPrice']?.toDouble() ?? 0.0,
      appFee: json['appFee']?.toDouble() ?? 0.0,
      discountAmount: json['discountAmount']?.toDouble() ?? 0.0,
      total: json['total']?.toDouble() ?? 0.0,
      method: json['method'],
      status: json['status'],
      currency: json['currency'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'billId': billId,
      'servicePrice': servicePrice,
      'distancePrice': distancePrice,
      'extraPrice': extraPrice,
      'totalPrice': totalPrice,
      'appFee': appFee,
      'discountAmount': discountAmount,
      'total': total,
      'method': method,
      'status': status,
      'currency': currency,
    };
  }
}

File: ResQ-Backend/src/main/java/com/livewithoutthinking/resq/entity/Bill.java
package com.livewithoutthinking.resq.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.Date;

@Entity
@Data
@Table(name = "bill")
public class Bill {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "BillID")
    private int billId;

    @OneToOne
    @JoinColumn(name = "rrid")
    private RequestRescue requestRescue;

    @Column(name = "ServicePrice")
    private double servicePrice; // Giá dịch vụ
    
    @Column(name = "DistancePrice")
    private double distancePrice; // Giá dịch vụ theo khoảng cách

    @Column(name = "ExtraPrice")
    private double extraPrice; // Giá dịch vụ phụ (nếu có)

    @Column(name = "TotalPrice")
    private double totalPrice; // Tổng số tiền cần thanh toán
    
    @ManyToOne
    @JoinColumn(name = "PaymentID", referencedColumnName = "PaymentID")
    private Payment payment; // Liên kết với bảng Payment

    @Column(name = "AppFee")
    private double appFee; // Giá dịch vụ theo loại

    @Column(name = "DiscountAmount")
    private double discountAmount; // Số tiền giảm
    
    @Column(name = "Total")
    private double total; // Giá đã được giảm

    @Column(name = "Method")
    private String method;

    @Column(name = "Status")
    private String status;

    @Column(name = "Currency")
    private String currency; // "VND" hoặc "USD"

    @Column(name = "Created_at", updatable = false)
    @Temporal(TemporalType.TIMESTAMP)
    private LocalDateTime createdAt; // Thời gian tạo

    @Column(name = "Updated_at")
    @Temporal(TemporalType.TIMESTAMP)
    private LocalDateTime updatedAt; // Thời gian cập nhật
}
